sources/web/datalab/settings.ts

```diff
@@ -116,16 +116,24 @@ export function getUserConfigDir(userId: string): string {
 /**
  * Copies the default user settings into the user's directory.
  */
-function copyDefaultUserSettings(userId: string) {
-  var userSettingsPath = path.join(getUserConfigDir(userId), SETTINGS_FILE);
+function getDefaultUserSettings(userId: string) {
   const defaultUserSettingsPath = path.join(__dirname, 'config', DEFAULT_USER_SETTINGS_FILE);
-  _log('Copying default settings: ' + defaultUserSettingsPath + ' to: ' + userSettingsPath);
+  _log('Getting default settings: ' + defaultUserSettingsPath);
   // Copy the default user settings file into user's directory.
   const defaultUserSettings = fs.readFileSync(defaultUserSettingsPath, {encoding: 'utf8'});
   const initialUserSettings = process.env.DATALAB_INITIAL_USER_SETTINGS;
   const mergedUserSettings : string = initialUserSettings ?
       mergeUserSettings(defaultUserSettings, initialUserSettings) : defaultUserSettings;
-  fs.writeFileSync(userSettingsPath, mergedUserSettings);
+  return mergedUserSettings;
+}
+
+/**
+ * Copies the default user settings into the user's directory.
+ */
+function copyDefaultUserSettings(userId: string) {
+  var userSettingsPath = path.join(getUserConfigDir(userId), SETTINGS_FILE);
+  _log('Copying default settings to: ' + userSettingsPath);
+  fs.writeFileSync(userSettingsPath, getDefaultUserSettings(userId));
```

Comment:
If this can now throw, we should either handle it here or make sure all callers catch. For example, the update function also tries to copy the settings file to the user's directory, which will effectively crash the server if the disk isn't mounted yet.

Do we want to support the same functionality in the case where the disk is mounted at an arbitrary point in time, or do we just want the server not to fail, but it's ok if we have a more basic behavior?

Reply:
It doesn't throw anything new.  We're just catching an existing failure below.

It's okay to have more basic behavior.

Reply:
Oh, I see the other use, catching there, too.